validator/src/main/java/com/google/daq/mqtt/sequencer/SequenceRunner.java

```diff
@@ -167,4 +165,35 @@ private void process(List<String> targetMethods) {
     });
   }
 
+  private List<String> getRunMethods(Class<?> clazz) {
+    return Arrays.stream(clazz.getMethods()).filter(this::shouldProcessMethod).map(Method::getName)
+        .filter(this::isTargetMethod).collect(Collectors.toList());
+  }
+
+  private boolean isTargetMethod(String methodName) {
+    return targets.isEmpty() || targets.contains(methodName);
+  }
+
+  private boolean shouldProcessMethod(Method method) {
+    Test test = method.getAnnotation(Test.class);
+    if (test == null) {
+      return false;
+    }
+    // If the target is explicitly indicated, then we should test it regardless of annotation.
+    if (targets.contains(method.getName())) {
+      return true;
+    }
+    Feature annotation = method.getAnnotation(Feature.class);
+    Stage stage = annotation == null ? Feature.DEFAULT_STAGE : annotation.stage();
+    return stage.processGiven(getDesiredStability());
+  }
+
+  private Stage getDesiredStability() {
+    String stability = SequenceBase.validatorConfig.stability;
+    return Strings.isNullOrEmpty(stability) ? DEFAULT_STABILITY : Stage.valueOf(stability);
```

Comment:
This stage vs stability naming does not seem congruent to me. 
I think we should pick one term, and it should be related to the enum decl. 
I think stage is better. We are choosing to test at a min or max threshold or level of stage. Alpha tests are in a stage of alpha. They might be quite stable, just preliminary. 
WDYT

Reply:
How about "feature_stage" -- same as Feature.Stage (as a class). I panicked and put "stability" because I didn't just want to use "stage" (too generic in the wider context) -- but I agree that's wasn't congruent.

Reply:
Relating to stage would be great. What about min_stage or something? I'm actually not wildly happy about min_ but it at least implies we are filtering from a range, not matching.